File: Problems & Solutions/1207. Unique Number of Occurrences/30 November 2022/Approach 2 - HashMap & HashSet/Unique Number of Occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> freq = new HashMap<>();
        for (int num : arr) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }
        
        Set<Integer> freqSet = new HashSet<>(freq.values());
        
        return freq.size() == freqSet.size();
    }
}

File: Problems & Solutions/1207. Unique Number of Occurrences/30 November 2022/Approach 2 - HashMap & HashSet/Unique Number of Occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> freq;
        for (int num : arr) {
            freq[num]++;
        }
        
        unordered_set<int> freqSet;
        for (auto [key, value] : freq) {
            freqSet.insert(value);
        }
        
        return freqSet.size() == freq.size();
    }
};